src/dfx/src/commands/ledger/mod.rs

```diff
@@ -117,15 +114,16 @@ pub async fn transfer(
     fee: ICPTs,
     from_subaccount: Option<Subaccount>,
     to: AccountIdBlob,
+    created_at_time: Option<u64>,
 ) -> DfxResult<BlockHeight> {
-    let timestamp_nanos = SystemTime::now()
-        .duration_since(UNIX_EPOCH)
-        .unwrap()
-        .as_nanos() as u64;
-
-    let mut retry_policy = ExponentialBackoff::default();
-
-    let block_height: BlockHeight = loop {
+    let timestamp_nanos = created_at_time.unwrap_or(
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_nanos() as u64,
+    );
+
+    let block_height: BlockHeight = {
```

Comment:
This `loop` should remain, as well as the retry policy and retry handling in this method.  Notice that even in the absence of a `--created-at-time` CLI parameter, dfx passes passes a `created_at` timestamp and attempts retries in an idempotent manner.

```suggestion
    );

    let mut retry_policy = ExponentialBackoff::default();

    let block_height: BlockHeight = loop {
```